frontend/app/components/date-of-birth.tsx

```diff
@@ -0,0 +1,42 @@
+import { InputField } from './input-field';
+import { InputSelect } from './input-select';
+
+export interface DateOfBirthProps {
+  lang: string;
+  legend: string;
+  monthLabel: string;
+  dayLabel: string;
+  yearLabel: string;
+  monthDefault?: number;
+  dayDefault?: number;
+  yearDefault?: number;
+  errorMessageMonth?: string;
+  errorMessageDay?: string;
+  errorMessageYear?: string;
+}
+
+export function DateOfBirth(props: DateOfBirthProps) {
+  const { lang, legend, monthLabel, dayLabel, yearLabel, monthDefault, dayDefault, yearDefault, errorMessageMonth, errorMessageDay, errorMessageYear } = props;
+  const monthOptions = Array.from({ length: 12 }, (_, i) => ({ children: new Intl.DateTimeFormat(`${lang}-ca`, { month: 'long' }).format(new Date(1900, i, 1)), value: i, id: `month-${i}` }));
+
+  return (
+    <fieldset>
+      <legend className="mb-2 font-semibold">{legend}</legend>
+      <div className="flex flex-col gap-6 sm:flex-row">
+        {lang == 'fr' && (
+          <>
+            <InputField id="day" label={dayLabel} name="day" inputMode="numeric" min={1} max={31} defaultValue={dayDefault} errorMessage={errorMessageDay} />
+            <InputSelect id="month" label={monthLabel} options={monthOptions} name="month" defaultValue={monthDefault} errorMessage={errorMessageMonth} />
+          </>
+        )}
+        {lang == 'en' && (
+          <>
+            <InputSelect id="month" label={monthLabel} options={monthOptions} name="month" defaultValue={monthDefault} errorMessage={errorMessageMonth} />
+            <InputField id="day" label={dayLabel} name="day" inputMode="numeric" min={1} max={31} defaultValue={dayDefault} errorMessage={errorMessageDay} />
+          </>
+        )}
+        <InputField id="year" label={yearLabel} name="year" inputMode="numeric" min={1900} defaultValue={yearDefault} errorMessage={errorMessageYear} />
```

Comment:
The `name="year"` attribute needs to be concatenated with the DateOfBirth component's name prop (doesn't exists at the moment) in scenarios where multiple instances of the DateOfBirth component are required on a single page. This comment goes for the other name attributres.

Reply:
any idea how to decouple the component from the page?  By concatenating a name to the respective fields I get issues with the validation.  I'm very confused how to solve this one.  I might need your help building this component out here because I'm feeling pretty stuck.

![image](https://github.com/DTS-STN/canadian-dental-care-plan/assets/48450599/08945e9a-a738-4cd5-8562-50d439ca0ee4)

Reply:
Can we allow the consumer of the component specify the `name` fields instead?

Reply:
I believe it's more practical not to validate date components (such as month, day, and year) individually. Let me explore alternative options. Perhaps storing the input values concatenated in a hidden field would be a more logical approach. Validating the complete date format (year-month-day) instead of separate validations seems more sensible to me, in my opinion.

Reply:
ha I just closed my other older PR that does full date validation.  Based on Figma, and GC Design System figma, they're validating individual date fields, which does seem a lot like overkill.  It's a new day today, maybe I can get this validation working :shrug: